integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -36,6 +37,33 @@ fn create_new_shard_layout(base_shard_layout: &ShardLayout, boundary_account: &s
     ShardLayout::v2(boundary_accounts, shard_ids.clone(), Some(shards_split_map))
 }
 
+use near_primitives_core::account::id::ParseAccountError;
+
+fn derive_shard_layout(
+    base_layout: &ShardLayout,
+    new_boundary_account: &str,
+) -> Result<ShardLayout, ParseAccountError> {
+    let mut boundary_accounts = base_layout.boundary_accounts().clone();
+    let mut shard_ids = base_layout.shard_ids().collect::<Vec<_>>();
+    let mut shards_split_map =
+        shard_ids.iter().map(|id| (*id, vec![*id])).collect::<BTreeMap<ShardId, Vec<ShardId>>>();
+
+    let new_boundary_account = new_boundary_account.parse::<AccountId>()?;
+    boundary_accounts.push(new_boundary_account.clone());
+    boundary_accounts.sort();
+    let new_boundary_account_index = boundary_accounts
+        .iter()
+        .position(|acc| acc == &new_boundary_account)
+        .expect("account should be ensured the exist");
+
+    let max_shard_id = *shard_ids.iter().max().expect("there should always be at lease one shard");
+    let new_shards = vec![max_shard_id + 1, max_shard_id + 2];
+    shard_ids.splice(new_boundary_account_index..new_boundary_account_index, new_shards.clone());
+    shards_split_map.insert(new_boundary_account_index.into(), new_shards);
```

Comment:
nice

Isn't `new_boundary_account_index..new_boundary_account_index` an empty range?

Reply:
Yeah, fixed. I got the right idea but execute it wrong.